### PR TITLE
[keyboard] Add rartpad

### PR DESCRIPTION
**Pull request by ladbahuy**
## Description

re-add new keyboard rartpad from #9410 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rart/rartpad/ratpad.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (zvecr) [inline keyboards/rart/rartpad/ratpad.h:1]: This file needs to be renamed `rartpad.h`

File: keyboards/rart/rartpad/ratpad.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* Copyright 2020 Alabahuy
```
Comment (zvecr) [inline keyboards/rart/rartpad/ratpad.c:1]: This file needs to be renamed `rartpad.c`

Comment (ladbahuy) [reply keyboards/rart/rartpad/ratpad.c]: done

File: keyboards/rart/rartpad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		MO(1),   KC_0,    KC_PDOT, KC_PENT
+	),
+  
+	[1] = LAYOUT_ortho_5x4(
+		KC_TRNS, RGB_HUI, RGB_HUD,  KC_TRNS, 
+		RGB_SAI, RGB_SAD, KC_MNXT,  KC_MPRV, 
+		RGB_VAI, RGB_VAD, KC_MSTP,  KC_MPLY, 
+		KC_COPY, KC_PSTE, KC_MYCM,  KC_CALC, 
+		KC_TRNS, RGB_TOG, KC_TRNS,  KC_TRNS 
+	)
```
Comment (Erovia) [inline keyboards/rart/rartpad/keymaps/default/keymap.c:36]:
```suggestion
    [0] = LAYOUT_ortho_5x4(
        KC_NLCK, KC_PSLS, KC_PAST, KC_INS,
        KC_P7,   KC_P8,   KC_P9,   KC_PEQL,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,   KC_PMNS,
        MO(1),   KC_0,    KC_PDOT, KC_PENT
    ),
  
    [1] = LAYOUT_ortho_5x4(
        KC_TRNS, RGB_HUI, RGB_HUD,  KC_TRNS, 
        RGB_SAI, RGB_SAD, KC_MNXT,  KC_MPRV, 
        RGB_VAI, RGB_VAD, KC_MSTP,  KC_MPLY, 
        KC_COPY, KC_PSTE, KC_MYCM,  KC_CALC, 
        KC_TRNS, RGB_TOG, KC_TRNS,  KC_TRNS 
    )
```

File: keyboards/rart/rartpad/keymaps/numpad/keymap.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_numpad_5x4(
+		KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
+		KC_P7,   KC_P8,   KC_P9,    
+		KC_P4,   KC_P5,   KC_P6,   KC_PPLS,         
+		KC_P1,   KC_P2,   KC_P3,    
+		KC_P0,            KC_PDOT,  KC_PENT         
+	)
```
Comment (Erovia) [inline keyboards/rart/rartpad/keymaps/numpad/keymap.c:28]:
```suggestion
    [0] = LAYOUT_numpad_5x4(
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
        KC_P7,   KC_P8,   KC_P9, 
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS, 
        KC_P1,   KC_P2,   KC_P3, 
        KC_P0,            KC_PDOT,  KC_PENT 
    )
```

File: keyboards/rart/rartpad/rartpad.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	K20, K21, K22, K13, \
+	K30, K31, K32,      \
+	K40,      K42, K33 \
+) \
+{ \
+	{ K00,   K01,   K02,   K03   }, \
+	{ K10,   K11,   K12,   K13   }, \
+	{ K20,   K21,   K22,   KC_NO }, \
+	{ K30,   K31,   K32,   K33   }, \
+	{ K40,   KC_NO, K42,   KC_NO }, \
```
Comment (Erovia) [inline keyboards/rart/rartpad/rartpad.h:40]:
```suggestion
    K00, K01, K02, K03, \
    K10, K11, K12, K13, \
    K20, K21, K22, K23, \
    K30, K31, K32, K33, \
    K40, K41, K42, K43  \
) \
{ \
    { K00,   K01,   K02,   K03 }, \
    { K10,   K11,   K12,   K13 }, \
    { K20,   K21,   K22,   K23 }, \
    { K30,   K31,   K32,   K33 }, \
    { K40,   K41,   K42,   K43 }, \
}
#define LAYOUT_numpad_5x4( \
    K00, K01, K02, K03, \
    K10, K11, K12,      \
    K20, K21, K22, K13, \
    K30, K31, K32,      \
    K40,      K42, K33 \
) \
{ \
    { K00,   K01,   K02,   K03   }, \
    { K10,   K11,   K12,   K13   }, \
    { K20,   K21,   K22,   KC_NO }, \
    { K30,   K31,   K32,   K33   }, \
    { K40,   KC_NO, K42,   KC_NO }, \
```

File: keyboards/rart/rartpad/readme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# [RARTPAD](https://github.com/alabahuy/RART/tree/master/RARTPAD)
+
```
Comment (Erovia) [inline keyboards/rart/rartpad/readme.md:2]:
```suggestion
# [RARTPAD](https://github.com/alabahuy/RART/tree/master/RARTPAD)

![RARTPAD](https://user-images.githubusercontent.com/30220306/84770444-c2ed2a00-b001-11ea-9f7b-f532ec736083.png)
```

Comment (Erovia) [reply keyboards/rart/rartpad/readme.md:2]: Unless you have a better picture of the board.

Comment (ladbahuy) [reply keyboards/rart/rartpad/readme.md:2]: done

**Comment by ladbahuy** [comment]
hay @zvecr, can it be approved?

File: keyboards/rart/rartpad/keymaps/numpad/keymap.c [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+bool led_update_user(led_t led_state) {
+    // writePin sets the pin high for 1 and low for 0.
+    // In this example the pins are inverted, setting
+    // it low/0 turns it on, and high/1 turns the LED off.
+    // This behavior depends on whether the LED is between the pin
+    // and VCC or the pin and GND.
+    writePin(NUM_LOCK_LED_PIN, led_state.num_lock);
+    
+    return false;
+
```
Comment (fauxpark) [inline keyboards/rart/rartpad/keymaps/numpad/keymap.c:40]:
```suggestion
```

File: keyboards/rart/rartpad/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+bool led_update_user(led_t led_state) {
+    // writePin sets the pin high for 1 and low for 0.
+    // In this example the pins are inverted, setting
+    // it low/0 turns it on, and high/1 turns the LED off.
+    // This behavior depends on whether the LED is between the pin
+    // and VCC or the pin and GND.
+    writePin(NUM_LOCK_LED_PIN, led_state.num_lock);
+    
+    return false;
+
```
Comment (fauxpark) [inline keyboards/rart/rartpad/keymaps/default/keymap.c:48]:
```suggestion
```

File: keyboards/rart/rartpad/config.h [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+#define MATRIX_COL_PINS { B2, D1, D2, D3 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#define NUM_LOCK_LED_PIN D7
+#define BACKLIGHT_LEVELS 0
```
Comment (fauxpark) [inline keyboards/rart/rartpad/config.h:45]:
```suggestion
```

File: keyboards/rart/rartpad/config.h [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+/* key matrix pins */
+#define MATRIX_ROW_PINS { B6, F6, D0, D4, C6 }
+#define MATRIX_COL_PINS { B2, D1, D2, D3 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
```
Comment (fauxpark) [inline keyboards/rart/rartpad/config.h:43]:
```suggestion
```

Comment (ladbahuy) [reply keyboards/rart/rartpad/config.h]: done @fauxpark

**Comment by alabahuy** [comment]
hay @fauxpark , can it be approved?

**Comment by ladbahuy** [comment]
hay @fauxpark , can it be approved?

File: keyboards/rart/rartpad/info.json [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+                {"label": "5", "x": 1, "y": 2},
+                {"label": "6", "x": 2, "y": 2},
+                
+                {"label": "1", "x": 0, "y": 3},
+                {"label": "2", "x": 1, "y": 3},
+                {"label": "3", "x": 2, "y": 3},
+                {"label": "Enter", "x": 3, "y": 3, "h": 2},
+                
+                {"label": "0", "x": 0, "y": 4, "w": 2},
+                {"label": ".", "x": 2, "y": 4}
```
Comment (fauxpark) [inline keyboards/rart/rartpad/info.json:58]:
```suggestion
                
                {"label": "4", "x": 0, "y": 2},
                {"label": "5", "x": 1, "y": 2},
                {"label": "6", "x": 2, "y": 2},
                {"label": "+", "x": 3, "y": 1, "h": 2},
                
                {"label": "1", "x": 0, "y": 3},
                {"label": "2", "x": 1, "y": 3},
                {"label": "3", "x": 2, "y": 3},
                
                {"label": "0", "x": 0, "y": 4, "w": 2},
                {"label": ".", "x": 2, "y": 4},
                {"label": "Enter", "x": 3, "y": 3, "h": 2}
```

Comment (ladbahuy) [reply keyboards/rart/rartpad/info.json]: done @fauxpark

**Comment by ladbahuy** [comment]
@Erovia can  it be merged?

**Comment by Erovia** [comment]
Yep.
Thank you for your contribution.

**Comment by ladbahuy** [comment]
yup, thanks before @Erovia

**Pull request merged**